docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,236 @@
+[id="{p}-troubleshooting"]
+== Troubleshoot a failing cluster
+
+When things don't seem to work as expected, there are several places you can look at to investigate.
+
+[float]
+[id="{p}-get-resources"]
+=== Get the list of resources
+
+To deploy and manage the Elastic stack, ECK creates several resources in the namespace the main resource is deployed in.
+
+For example, each Elasticsearch node and Kibana instance has a dedicated Pod.
+Check the Pods running and their status, to compare with the expected instances:
+
+[source,sh]
+----
+kubectl get pods
+
+NAME                                 READY     STATUS    RESTARTS   AGE
+elasticsearch-sample-es-66sv6dvt7g   0/1       Pending   0          3s
+elasticsearch-sample-es-9xzzhmgd4h   1/1       Running   0          27m
+elasticsearch-sample-es-lgphkv9p67   0/1       Pending   0          3s
+kibana-sample-kb-5468b8685d-c7mdp    0/1       Running   0          4s
+----
+
+Check the services:
+
+[source,sh]
+----
+kubectl get services
+
+NAME                           TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+elasticsearch-sample-es-http   ClusterIP   10.19.248.93    <none>        9200/TCP   2d
+kibana-sample-kb-http          ClusterIP   10.19.246.116   <none>        5601/TCP   3d
+----
+
+[float]
+[id="{p}-describe-failing-resources"]
+=== Describe failing resources
+
+If an Elasticsearch node does not come up, it is possible the associated Pod cannot be scheduled by Kubernetes.
+You may see this by looking at the Status of the Pods. If it does not become `Running` after a few seconds, chances are something's wrong:
+
+[source,sh]
+----
+kubectl get pods
+
+NAME                                 READY     STATUS    RESTARTS   AGE
+elasticsearch-sample-es-66sv6dvt7g   0/1       Pending   0          3s
+elasticsearch-sample-es-9xzzhmgd4h   1/1       Running   0          27m
+elasticsearch-sample-es-lgphkv9p67   0/1       Pending   0          3s
+kibana-sample-kb-5468b8685d-c7mdp    0/1       Running   0          4s
+----
+
+Get more details about the reason `elasticsearch-sample-es-lgphkv9p67` isn't scheduled:
+
+[source,sh]
+----
+kubectl describe pod elasticsearch-sample-es-lgphkv9p67
+
+(...)
+Events:
+  Type     Reason             Age               From                Message
+  ----     ------             ----              ----                -------
+  Warning  FailedScheduling   1m (x6 over 1m)   default-scheduler   pod has unbound immediate PersistentVolumeClaims (repeated 2 times)
+  Warning  FailedScheduling   1m (x6 over 1m)   default-scheduler   pod has unbound immediate PersistentVolumeClaims
+  Warning  FailedScheduling   1m (x11 over 1m)  default-scheduler   0/3 nodes are available: 1 node(s) had no available volume zone, 2 Insufficient memory.
+  Normal   NotTriggerScaleUp  4s (x11 over 1m)  cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added)
+----
+
```

Comment:
"If you see an error with unbound persistent volume claims (PVCs), it means there is not currently a persistent volume that can satisfy the claim. If you are using automatically provisioned storage (e.g. Amazon EBS provisioner), sometimes the storage provider can take a few minutes to provision a volume, so this may resolve itself in a few minutes. You can also check the status by running `kubectl describe persistentvolumeclaims` to see events of the PVCs."